### PR TITLE
Provide Idiomatically Cased Pay ID Classes

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

JavaScript tends to favor only casing the first letter of acronyms (PayId vs PayID). Start to re-case our classes to this style. 

### Context of Change

We use both styles of this casing across our JavaScript libraries (see `IlpClient` and `XRPClient`). JavaScript tends to favor the former case. 

This PR re-cases the Pay ID functionality to use the correct casing. Specifically:
- Duplicate each incorrectly cased class into a correctly cased class. 
- Modify the incorrectly cased class to call the correctly case classed
- Mark the incorrectly cased class as deprecated
- Update our tests immediately to the new casing
- Update `index.ts` to export both classes. 

A future PR will remove the incorrectly cased classes in N+2 releases, as described by our new release policy. 

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

A bridge to correctly named classes is provided. 

## Test Plan

CI

### CODE REVIEW

File: src/PayID/pay-id-components.ts [hunk 1]
```diff
@@ -1,7 +1,29 @@
+// Disable multiple classes to accommodate the switch to idiomatic style naming.
+// TODO(keefertaylor): Remove this when migration is complete.
+/* eslint-disable  max-classes-per-file */
```
Comment (hbergren) [inline src/PayID/pay-id-components.ts:3]: Super helpful comment for this `eslint-disable`.

File: src/PayID/pay-id-utils.ts [hunk 2]
```diff
@@ -49,3 +53,24 @@ export default class PayIDUtils {
...
+   * @param parsePayID - The input Pay ID.
+   * @returns A PayIDComponents object if the input was valid, otherwise undefined.
+   */
+  public static parsePayID(payID: string): PayIDComponents | undefined {
+    return PayIdUtils.parsePayID(payID)
+  }
+
+  /** Please do not instantiate this static utility class. */
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  private constructor() {}
```
Comment (hbergren) [inline src/PayID/pay-id-utils.ts:75]: Good comment.

Total aside: There's a `@typescript-eslint` rule that disallows using "classes as namespaces", which a class with only static methods is. I know that's a pretty common pattern in `xpring-common-js`, and I just wanted to get a feel for how you felt about that.

[Here is the actual rule documentation for some reasoning and examples](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-extraneous-class.md).

File: src/PayID/pay-id-utils.ts [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
+import PayIdComponents, { PayIDComponents } from './pay-id-components'
 
 /**
- * A static utility class for PayID.
+ * A static utility class for Pay ID functionality.
  */
-export default class PayIDUtils {
+export default class PayIdUtils {
   /**
    * Parse a PayID string to a set of PayIDComponents object
```
Comment (amiecorso) [inline src/PayID/pay-id-utils.ts:12]: should this line be `PayIdComponents` ?

File: src/PayID/pay-id-utils.ts [hunk 4]
```diff
@@ -1,17 +1,21 @@
...
    *
    * @param parsePayID - The input Pay ID.
    * @returns A PayIDComponents object if the input was valid, otherwise undefined.
```
Comment (amiecorso) [inline src/PayID/pay-id-utils.ts:15]: here too

File: test/PayID/pay-id-utils.test.ts [hunk 5]
```diff
@@ -11,7 +11,7 @@ describe('PayIDUtils', function (): void {
     const rawPayID = `${path}$${host}`
 
     // WHEN it is parsed to components.
-    const payIDComponents = PayIDUtils.parsePayID(rawPayID)
+    const payIDComponents = PayIdUtils.parsePayID(rawPayID)
```
Comment (amiecorso) [inline test/PayID/pay-id-utils.test.ts:14]: does it matter that this `payIDComponents` variable is cased in the old style?  I know the style applies mostly to classnames but wondering if we would benefit from that consistency wherever possible?

Comment (amiecorso) [reply test/PayID/pay-id-utils.test.ts:14]: (here and throughout the file)

**Comment by hbergren** [comment]
Random question, are `XRP` class and variable names still going to be in full cap casing, or will they be `Xrp` eventually?

**Comment by keefertaylor** [comment]
> Random question, are XRP class and variable names still going to be in full cap casing, or will they be Xrp eventually?

We talked offline. Aiming for Xrp since we can enforce with tooling.

**Pull request merged**